src/Layer.js

```diff
@@ -0,0 +1,67 @@
+import cx from 'classnames'
+import React, { createContext, useState, useContext } from 'react'
+import { createPortal } from 'react-dom'
+import propTypes from '@dhis2/prop-types'
+import { oneOf } from 'prop-types'
+
+const LayerContext = createContext({
+    node: document.body,
+    level: 0,
+})
+
+const Layer = ({ children, className, level, position }) => {
+    const parentLayer = useContext(LayerContext)
+    const [layerEl, setLayerEl] = useState(null)
+    const nextLayer = {
+        node: layerEl,
+        level: Math.max(parentLayer.level, level),
+    }
+    const portalNode =
+        level < parentLayer.level ? parentLayer.node : document.body
+
+    return createPortal(
+        <div
+            ref={setLayerEl}
+            className={cx(className, position, `level-${level}`)}
+        >
+            {layerEl && (
+                <LayerContext.Provider value={nextLayer}>
+                    {children}
+                </LayerContext.Provider>
+            )}
+            <style jsx>{`
+                div {
+                    z-index: ${level};
+                    pointer-events: none;
```

Comment:
Doesn't this stop all pointer events from reaching the children? Also I don't understand why we need this here?

Reply:
Upon reflection, I should have removed that line. I have some ideas about how we could have blocking VS non-blocking layers and use pointer-events for that. But for now that's out of scope and just causes confusion. 

I've deleted it in https://github.com/dhis2/ui-core/pull/807/commits/a2286ba0ea4a0668e31c2378d51a679588f7ef3c.